calibration_utils.py

```diff
@@ -554,6 +581,41 @@ def calibrate_camera_charuco(self, allCorners, allIds, imsize, hfov):
         if self.traceLevel == 3 or self.traceLevel == 10:
             print('Per View Errors...')
             print(perViewErrors)
+        # check if there are any suspicious corners with high reprojection error
+        corners_removed = False
+        for i in range(len(perViewErrors)):
+            if perViewErrors[i] > 3:
+                corners = allCorners[i]
+                ids = allIds[i]
+                objpts = self.charuco_ids_to_objpoints(ids)
+                errs = self.compute_reprojection_errors(objpts, corners, camera_matrix, distortion_coefficients, rotation_vectors[i], translation_vectors[i])
+                
+                suspicious_err_thr = max(2*np.median(errs), 100)
+                n_offending_pts = np.sum(errs > suspicious_err_thr)
+                offending_pts_idxs = np.where(errs > suspicious_err_thr)[0]
+                # check if there are offending points and if they form a minority
+                if n_offending_pts > 0 and n_offending_pts < len(corners)/5:
+                    print(f"removing {n_offending_pts} offending points with errs {errs[offending_pts_idxs]}")
+                    corners_removed = True
+                    #remove the offending points
+                    offset = 0
+                    allCorners[i] = np.delete(allCorners[i],offending_pts_idxs, axis = 0)
+                    allIds[i] = np.delete(allIds[i],offending_pts_idxs, axis = 0)
+        if corners_removed:
+            # recompute the calibration if we removed any offending points
+            print("recomputing intrinsics")
+            (ret, camera_matrix, distortion_coefficients,
+                rotation_vectors, translation_vectors,
+                stdDeviationsIntrinsics, stdDeviationsExtrinsics,
+                perViewErrors) = cv2.aruco.calibrateCameraCharucoExtended(
```

Comment:
Can we add additonal condition where the there is good amount of corners are still available after removing the bad ones.

Reply:
Good point, there is already a check to not remove any corners if they are more than 1/5 of the total corners. This should be enough to never have less than 4 corners which is identical to previous checks, right?